src/cargo/core/shell.rs

```diff
@@ -565,6 +579,21 @@ fn supports_unicode(stream: &dyn IsTerminal) -> bool {
     !stream.is_terminal() || supports_unicode::supports_unicode()
 }
 
+/// Detects if the terminal supports OSC 9;4 progress notifications.
+#[allow(clippy::disallowed_methods)] // ALLOWED: to read terminal app signature
+fn supports_progress_report() -> bool {
+    // Windows Terminal session
+    std::env::var("WT_SESSION").is_ok()
+    // Compatibility with ConEmu's ANSI support
+        || std::env::var("ConEmuANSI").ok() == Some("ON".into())
+    // WezTerm: check if TERM_PROGRAM is "WezTerm" and the version is recent enough
+    // https://github.com/rust-lang/cargo/pull/14615#issuecomment-2646738819
+    || (std::env::var("TERM_PROGRAM").ok() == Some("WezTerm".into())
+        && std::env::var("TERM_PROGRAM_VERSION")
+            .ok()
+            .map_or(false, |v| v.as_str() >= "20250209-182623-44866cc1"))
```

Comment:
I think the main reason for version checks is to deal with buggy terminal implementations.  If older wezterms gracefully handle these codes, should we skip the version check?

Reply:
FWIW, older versions of wezterm will swallow an OSC they doesn't understand, so you could remove the version number check and just check for wezterm without it.  That feels like less magic to keep track of over here.

Reply:
I've removed the version check for WezTerm in the latest edit.